src/ArchiveFolder.cs

```diff
@@ -74,25 +83,19 @@ public async Task<IChildFile> CreateFileAsync(string name, bool overwrite = fals
 
     protected override ReadOnlyArchiveFolder WrapSubfolder(string name) => new ArchiveFolder(this, name);
 
-    protected async Task RemoveSubfolder(IArchiveEntry entry, CancellationToken cancellationToken = default)
-    {
-        var archive = await OpenWritableArchiveAsync(cancellationToken);
-        archive.RemoveEntry(entry);
-
-        var subfolders = await GetSubfoldersAsync(cancellationToken);
-        subfolders.Remove(entry.Key);
-    }
-
     protected async Task RemoveSubfolder(string key, CancellationToken cancellationToken = default)
     {
         cancellationToken.ThrowIfCancellationRequested();
-
+        
+        // Remove this and all child entries from archive.
+        // Force enumeration with .ToList() since we're modifying the collection.
         var archive = await OpenWritableArchiveAsync(cancellationToken);
+        var entries = archive.Entries.ToList();
+        foreach (var entry in entries)
+            if (entry.Key == key || IsChild(entry.Key, key))
```

Comment:
The key change in here is that we remove *every* child entry, not just the requested folder.